vpr/src/util/vpr_utils.cpp

```diff
@@ -2021,3 +2021,15 @@ void print_timing_stats(std::string name,
             current.num_full_hold_updates - past.num_full_hold_updates,
             current.num_full_setup_hold_updates - past.num_full_setup_hold_updates);
 }
+
+void error_if_timing_failed() {
+    auto& timing_ctx = g_vpr_ctx.mutable_timing();
+    // Everytime a negative slack is found, it adds on to timing_ctx.fail_timing_msg where it failed.
+    // If timing_ctx.fail_timing_msg is empty, then no negative slacks were found.
+    if (timing_ctx.fail_timing_msg != "") {
```

Comment:
Seems like this could be localized to a timing_something.cpp file, instead of being put in a general utility file.  This probably isn't a utility that many will call.